src/scripts/gym/GymBattle.ts

```diff
@@ -13,6 +13,10 @@ class GymBattle extends Battle {
     public static defeatPokemon() {
         player.gainMoney(this.enemyPokemon().money);
         player.gainExp(this.enemyPokemon().exp, this.enemyPokemon().level, false);
+        player.gainShards(this.enemyPokemon().type1, 5);
+        if (this.enemyPokemon().type2 != -1) {
+            player.gainShards(this.enemyPokemon().type2, 5);
+        }
```

Comment:
Not sure what your opinion on this is, but maybe the `None` type check should be done in the player.gainShards method?